project-base/storefront/hooks/order/useChangePaymentInOrder.ts

```diff
@@ -0,0 +1,42 @@
+import { useChangePaymentInOrderMutationApi } from 'graphql/generated';
+import { getInternationalizedStaticUrls } from 'helpers/getInternationalizedStaticUrls';
+import { showSuccessMessage } from 'helpers/toasts';
+import { useIsUserLoggedIn } from 'hooks/auth/useIsUserLoggedIn';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import useTranslation from 'next-translate/useTranslation';
+import { useRouter } from 'next/router';
+
+export const useChangePaymentInOrder = () => {
+    const { t } = useTranslation();
+    const router = useRouter();
+    const isUserLoggedIn = useIsUserLoggedIn();
+    const { url } = useDomainConfig();
+    const [orderByHashUrl, customerOrderDetailUrl] = getInternationalizedStaticUrls(
+        [{ url: '/order-detail/:urlHash', param: '' }, '/customer/order-detail'],
+        url,
+    );
+
+    const [{ fetching: changePaymentInOrderFetching }, changePaymentInOrder] = useChangePaymentInOrderMutationApi();
+
+    const changePaymentInOrderHandler = async (orderUuid: string, paymentUuid: string) => {
+        const { data: changePaymentInOrderData } = await changePaymentInOrder({
+            input: { orderUuid, paymentGoPayBankSwift: null, paymentUuid },
+        });
+        const editedOrder = changePaymentInOrderData?.ChangePaymentInOrder;
+
+        if (editedOrder) {
+            showSuccessMessage(t('Your payment has been successfully changed'));
+
+            if (isUserLoggedIn) {
+                router.push({
+                    pathname: customerOrderDetailUrl,
+                    query: { orderNumber: editedOrder.number },
+                });
+            } else {
+                router.push(`${orderByHashUrl}/${editedOrder.urlHash}`);
```

Comment:
Isn't this URL relative? Won't that cause issues?

Reply:
Good question. I did a demo where I tested router.push on the /stores page
```
useEffect(() => {
    router.push('/order-url/test');
}, []);
```
And the resulting url was /order-url/test so I guess my solution ok :)